File: jobs/2026-05-30__07-57-35/encore_multi_service_api_ts__efof5d2/artifacts/user/myapp/order/order.ts
import { api } from "encore.dev/api";
import { user } from "~encore/clients";

interface OrderResponse {
    orderId: number;
    userId: number;
    userName: string;
}

export const getOrder = api(
    { expose: true, method: "GET", path: "/order/:id" },
    async ({ id }: { id: number }): Promise<OrderResponse> => {
        const u = await user.getUser({ id: 1 });
        return {
            orderId: Number(id),
            userId: 1,
            userName: u.name,
        };
    }
);

File: jobs/2026-07-03__03-23-57/encore_multi_service_api_go__8mKMTqK/artifacts/user/myproject/order/order.go
package order

import (
	"context"

	"encore.app/user"
)

type Order struct {
	ID       string `json:"id"`
	UserID   string `json:"user_id"`
	UserName string `json:"user_name"`
}

//encore:api public method=GET path=/order/:id
func GetOrder(ctx context.Context, id string) (*Order, error) {
	userId := "usr_" + id

	usr, err := user.GetUser(ctx, userId)
	if err != nil {
		return nil, err
	}

	return &Order{
		ID:       id,
		UserID:   usr.ID,
		UserName: usr.Name,
	}, nil
}
